include/CoreGraphics/CoreGraphicsExport.h

```diff
@@ -66,5 +67,94 @@ typedef struct __CGPath* CGMutablePathRef;
 // typedef struct __Foo* FooRef;
 // indiciating backing code need some clean up
 typedef void* CGDataProviderRef;
+
+#if defined(__clang__)
 typedef id CGFontRef;
-typedef id CGPatternRef;
\ No newline at end of file
+typedef id CGPatternRef;
+#endif
+
+typedef enum {
+    kCGImageAlphaNone,
+    kCGImageAlphaPremultipliedLast,
+    kCGImageAlphaPremultipliedFirst,
+    kCGImageAlphaLast,
+    kCGImageAlphaFirst,
+    kCGImageAlphaNoneSkipLast,
+    kCGImageAlphaNoneSkipFirst,
+    kCGImageAlphaOnly
+} CGImageAlphaInfo;
+enum {
+    kCGBitmapAlphaInfoMask = 0x1F,
+    kCGBitmapFloatComponents = (1 << 8),
+    kCGBitmapByteOrderMask = 0x7000,
+    kCGBitmapByteOrderDefault = (0 << 12),
+    kCGBitmapByteOrder16Little = (1 << 12),
+    kCGBitmapByteOrder32Little = (2 << 12),
+    kCGBitmapByteOrder16Big = (3 << 12),
+    kCGBitmapByteOrder32Big = (4 << 12)
+};
+typedef int CGBitmapInfo;
+
+typedef enum {
+    kCGRenderingIntentDefault,
+    kCGRenderingIntentAbsoluteColorimetric,
+    kCGRenderingIntentRelativeColorimetric,
+    kCGRenderingIntentSaturation,
+    kCGRenderingIntentPerceptual,
+} CGColorRenderingIntent;
+
+typedef enum {
+    kCGColorSpaceModelUnknown = -1,
+    kCGColorSpaceModelMonochrome,
+    kCGColorSpaceModelRGB,
+    kCGColorSpaceModelCMYK,
+    kCGColorSpaceModelLab,
+    kCGColorSpaceModelDeviceN,
+    kCGColorSpaceModelIndexed,
+    kCGColorSpaceModelPattern,
+} CGColorSpaceModel;
+
+typedef enum {
+    _Color565 = 0,
+    _ColorARGB,
+    _ColorABGR,
+    _ColorBGRX,
+    _ColorXBGR,
+    _ColorGrayscale,
+    _ColorBGR,
+    _ColorA8,
+    _ColorIndexed,
+    _ColorMax,
+    _ColorRGBA = _ColorABGR
+} surfaceFormat;
+
+typedef struct {
+    CGColorSpaceModel colorSpaceModel;
+    CGBitmapInfo bitmapInfo;
+    unsigned int bitsPerComponent;
+    unsigned int bytesPerPixel;
+} __CGFormatProperties;
```

Comment:
I kept them here because they are used internally by many different components and effectively allows other components to just import CoreGraphicsExport.h in order to make use of them. I can put them in a separate file if that works. Perhaps CoreGraphicsInternal.h?